homeassistant/components/izone/climate.py

```diff
@@ -0,0 +1,468 @@
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from ...const import (
+    ATTR_TEMPERATURE, PRECISION_HALVES, TEMP_CELSIUS,
+    STATE_OFF, STATE_ON)
+from ...helpers.temperature import display_temp as show_temp
+from ...helpers.typing import ConfigType, HomeAssistantType
+
+from .constants import DATA_ADD_ENTRIES, DATA_DISCOVERY_SERVICE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, config: ConfigType,
+                            async_add_entities):
+    """Initialize an IZone Controller."""
+    disco = hass.data[DATA_DISCOVERY_SERVICE]
+
+    # create any components not yet created
+    for controller in disco.controllers.values():
+        disco.init_controller(controller, async_add_entities)
+
+    # disco will use the register function to register any further components
+    hass.data[DATA_ADD_ENTRIES] = async_add_entities
+
+    return True
+
+
+class ControllerDevice(ClimateDevice):
+    """Representation of iZone Controller."""
+
+    def __init__(self, controller, async_add_entities) -> None:
+        """Initialise ControllerDevice."""
+        self._controller = cast(Controller, controller)
+
+        self._supported_features = (SUPPORT_OPERATION_MODE |
+                                    SUPPORT_FAN_MODE | SUPPORT_ON_OFF)
+
+        if ((controller.ras_mode == 'master' and controller.zone_ctrl == 13) or
+                controller.ras_mode == 'RAS'):
+            self._supported_features |= SUPPORT_TARGET_TEMPERATURE
+
+        self._state_to_pizone = {
+            STATE_COOL: Controller.Mode.COOL,
+            STATE_HEAT: Controller.Mode.HEAT,
+            STATE_AUTO: Controller.Mode.AUTO,
+            STATE_FAN_ONLY: Controller.Mode.VENT,
+            STATE_DRY: Controller.Mode.DRY,
+        }
+        if controller.free_air_enabled:
+            self._state_to_pizone[STATE_ECO] = Controller.Mode.FREE_AIR
+
+        self._fan_to_pizone = {}
+        for fan in controller.fan_modes:
+            self._fan_to_pizone[fan.name.title()] = fan
+
+        self._device_info = {
+            'identifiers': {
+                (DOMAIN, self.unique_id)
+            },
+            'name': self.name,
+            'manufacturer': 'IZone',
+            'model': self._controller.sys_type,
+        }
+        async_add_entities([self], True)
+
+        # Create the zones
+        self.zones = {}
+        for zone in controller.zones:
+            self.zones[zone] = ZoneDevice(self, zone)
+
+        async_add_entities(self.zones.values(), True)
+
+        self._available = True
+
+        _LOGGER.info("Controller UID=%s is ready", controller.device_uid)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return not self._available
```

Comment:
I don't think this makes a difference - if the entity is unavailable the state is not read out.